src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -1531,9 +1533,15 @@ subroutine ocn_compute_KPP_input_fields(statePool, forcingPool, meshPool, diagno
 
          nonLocalSurfaceTracerFlux(indexTempFlux, iCell) = activeTracersSurfaceFlux(indexTempFlux,iCell) &
                  + penetrativeTemperatureFlux(iCell) - penetrativeTemperatureFluxOBL(iCell)  &
-                 - fracAbsorbed * (rainFlux(iCell) + evaporationFlux(iCell)) * activeTracers(indexTempFlux,1,iCell)/rho_sw  &
-                 - fracAbsorbedRunoff * surfaceThicknessFluxRunoff(iCell) &
-                  * min(activeTracers(indexTempFlux,1,iCell),0.0_RKIND)/rho_sw
+                 - fracAbsorbed * (rainFlux(iCell) + evaporationFlux(iCell)) * activeTracers(indexTempFlux,1,iCell)/rho_sw &
+                 - fracAbsorbedRunoff * surfaceThicknessFluxRunoff(iCell)* min(activeTracers(indexTempFlux,1,iCell),0.0_RKIND)/rho_sw
+
+         if(config_cvmix_use_kpp_buoyancyForcing_fix) then
```

Comment:
@philipwjones @mattdturner or @amametjanov:
We put this if flag here for backwards compatibility. It is within a cell loop. 

Is this still a performance sin, or do the compilers handle this now for flags that never change during the run?

Reply:
It's a sin my son and will require at least 3 Hail Marys. Particularly in this case, where the loop is complex enough that the compiler can't break it up, it's not optimal. I don't know that it'll be a huge performance hit on CPU architectures but will probably need to be broken out for the GPU version.

Reply:
this will hopefully be a relatively short term hack.  we just need to remember to remove it once we are satisfied with the new (correct) code.

Reply:
agreed.  Once we test this in WC and are happy with results, I'll make a PR to remove the if statement.  I can't imagine this is in for more than a week.

Reply:
Perfect, thanks for the perspective. I made an issue to help us remember: https://github.com/MPAS-Dev/MPAS-Model/issues/692